### PR TITLE
[4.2] fix var name in content table

### PR DESCRIPTION
**Pull request by heelc29**
Fix for Pull Request #37017 .

### Summary of Changes

fix variable name: content table use `state` instead of `published`

### Testing Instructions

- try to create two articles with the same alias
- the existing article is either published or trashed
- check the error messages

### Actual result BEFORE applying this Pull Request

error message `COM_CONTENT_ERROR_UNIQUE_ALIAS_TRASHED="A trashed Article in this category has the same alias."` will not appear

image

### Expected result AFTER applying this Pull Request

error message `COM_CONTENT_ERROR_UNIQUE_ALIAS_TRASHED="A trashed Article in this category has the same alias."` will appear

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [x] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [x] No documentation changes for manual.joomla.org needed

### CODE REVIEW

**Comment by alikon** [comment]
I have tested this item :white_check_mark: successfully on fa0c6e55d66dc7cef45e3d32d9ba9c6e9343fe27<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39138">issues.joomla.org/tracker/joomla-cms/39138</a>.</sub>

**Comment by viocassel** [comment]
I have tested this item :white_check_mark: successfully on fa0c6e55d66dc7cef45e3d32d9ba9c6e9343fe27<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39138">issues.joomla.org/tracker/joomla-cms/39138</a>.</sub>

**Comment by richard67** [comment]
RTC <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39138">issues.joomla.org/tracker/joomla-cms/39138</a>.</sub>

File: libraries/src/Table/Content.php [hunk 1]
```diff
@@ -349,7 +349,7 @@ public function store($updateNulls = true)
             // Is the existing article trashed?
             $this->setError(Text::_('COM_CONTENT_ERROR_UNIQUE_ALIAS'));
 
-            if ($table->published === -2) {
+            if ($table->state === -2) {
```
Comment (laoneo) [inline libraries/src/Table/Content.php:352]: Should we not use here `getColumnAlias` to be on the safe side?

Comment (fancyFranci) [reply libraries/src/Table/Content.php:352]: I decide to merge the bug fix without alias, so at least the message is working again. Further code improvement can happen in another PR.

**Comment by fancyFranci** [comment]
Always nice to get feedback from the system. Thank you!

**Pull request merged**